### PR TITLE
Builder override

### PR DESCRIPTION
**Pull request by yitzchak**
Allow `%!TEX builder = ...` magic to override builder.

### CODE REVIEW

**Comment by thomasjo** [comment]
It would be great if you could add a test (or two) for the new functionality. We need to specs to prevent regressions, etc.

**Comment by yitzchak** [comment]
Got it. I'll make it happen! :bulb:

**Comment by yitzchak** [comment]
I added a few tests, but I still have a few doubts about the functionality. I am planning on testing on some Windows machines next week.

File: spec/build-registry-spec.js [hunk 1]
```diff
@@ -32,10 +32,22 @@ describe('BuilderRegistry', () => {
       expect(registry.getBuilder(filePath).name).toEqual('TexifyBuilder')
     })
 
-    it('throws an error when unable to resolve ambigious builder registration', () => {
+    it('throws an error when unable to resolve ambiguous builder registration', () => {
```
Comment (thomasjo) [inline spec/build-registry-spec.js]: Nice catch! 😁

**Comment by yitzchak** [comment]
Okay, I am satisfied. I tested in on MiKTeX with both `texify` and `latexmk` installed and was able to switch back and forth between the two with the magic comment.

I did have to move the saving of the file in `Composer` to before `getBuilder` was called to ensure that `MagicParser` would see a changed builder override originating from an unsaved file.

**Comment by thomasjo** [comment]
LGTM! Feel free to hit that big, green button 👍

**Comment by yitzchak** [comment]
Awesome! :boom:

**Pull request merged**